### ISSUE TITLE
Cannot go from .to_dict() back to Js types.

### ISSUE THREAD

**Issue by Hucaru** [PiotrDabkowski/Js2Py#52]
I am using the esprima translated example provided and have translated escodegen to allow me to rebuild code from the AST. I have noticed the following issue occur for certain scripts:

`js2py.base.PyJsException: TypeError: Undefiend and null dont have properties!`

This only occurs when I convert the datatype returned by esprima to a dictionary. If I then pass that converted data (without modification) back to escodegen it will throw the error above. If I don't convert to a dictionary it runs just fine. This only occurs with certain scripts, for example:

```
define(["definition"], function (foo) {
    "use strict";

    return foo(null, {
        declaredClass: "className",

        constructor: function () {
          this.var = 1;
        },

        id: function () {
            return 1;
        },

        type: function () {
            return 1;
        },

        namespace: function () {
            return 1;
        }
    });
});
```

**Comment by PiotrDabkowski** [comment]
.to_dict() is including only enumerable properties of the object inside dict. So if some property is not enumerable it will not be included in the dict and if you convert that dict to JS it will be missing this property, hence the error.

**Comment by Hucaru** [comment]
Oh I see. However, in this scenario I cannot figure out what property is being lost. When comparing the Json generated from using esprima.js in a browser and the json representation of the dictionary generated by pyesprima no difference is present.

**Comment by PiotrDabkowski** [comment]
Is browser showing also non enumerable properties? I cant think of another reason for that error other than non enumerable properties. Do you maybe know what property failed to access?

**Comment by Hucaru** [comment]
Yes in the example provided it cannot process the null on the inheritance - `return foo(null, {`

**Comment by PiotrDabkowski** [comment]
Oh, I think I know what the reason might be. Both null and undefined are converted to None during to_dict(). Afterward when you convert that dict back, None will be converted to undefined. Maybe that is the reason. I dont have escodegen translated and I cant explore whats going on so its just a guess. I will check the details today evening when I have some time, now I am at work :D

**Comment by Hucaru** [comment]
Yes that would explain both the error message and why no difference appears when comparing the json data before and after dictionary conversion. Thanks for taking a look! Love the lib saved me many hours so far in manual translation :)
